sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -315,20 +324,35 @@ public Mono<HttpResponse> logResponse(ClientLogger logger, HttpResponseLoggingCo
             }
 
             LoggingEventBuilder logBuilder = getLogBuilder(logLevel, logger);
+
             addBasicResponseProperties(logger, loggingOptions, response, logBuilder);
+
             Long contentLength = getAndLogContentLength(response.getHeaders(), logBuilder, logger);
 
             if (httpLogDetailLevel.shouldLogBody()) {
                 String contentTypeHeader = response.getHeaderValue(HttpHeaderName.CONTENT_TYPE);
+
                 if (shouldBodyBeLogged(contentTypeHeader, contentLength)) {
                     // shouldBodyBeLogged ensures contentLength is not null and within limits.
                     int contentLengthInt = contentLength.intValue();
-                    return Mono.just(new LoggingHttpResponse(response, logBuilder, logger, contentLengthInt,
-                        contentTypeHeader, prettyPrintBody));
+
+                    AccessibleByteArrayOutputStream stream = new AccessibleByteArrayOutputStream(contentLengthInt);
+
+                    response.getBody().subscribe(byteBuffer -> {
```

Comment:
I'd recommend against using `subscribe` within a reactive flow anywhere that isn't a sink-based call chain. This doesn't block and may result in race conditions or unexpected behaviors.